### PR TITLE
NAS-105221 / 12.0 / Nas 105221

### PR DESCRIPTION
**Pull request by erinix**

### CODE REVIEW

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 1]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
 Override the following errors?"),
 force_warnings:{
     'diskSizeWarning': T("One or more data vdevs has disks of different sizes.")
-}
+},
+data_vdev_title: T("Data"),
+data_vdev_description: T("Normal vdev type, used for primary storage operations. A ZFS pool will always have at least one DATA vdev."),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts]:
```suggestion
data_vdev_description: T("Normal vdev type, used for primary storage operations. ZFS pools always have at least one DATA vdev."),
```

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 2]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
...
+cache_vdev_title: T("Cache"),
+cache_vdev_description: T("ZFS L2ARC read-cache. Can be removed. Optional vdev that can be used with fast devices to accelerate read operations."),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts]:
```suggestion
cache_vdev_description: T("ZFS L2ARC read-cache that can be used with fast devices to accelerate read operations. Optional vdev that can be removed."),
```

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 3]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
...
+log_vdev_title: T("Log"),
+log_vdev_description: T("ZFS LOG device. Can be removed. Optional write-cache that can improve speeds of synchronous writes."),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts]:
```suggestion
log_vdev_description: T("ZFS LOG device that can improve speeds of synchronous writes. Optional write-cache that can be removed."),
```

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 4]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
...
+spare_vdev_title: T("Hot Spare"),
+spare_vdev_description: T("Drive set aside ready to be inserted into DATA pool vdevs when a device has failed."),
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts]:
```suggestion
spare_vdev_description: T("Drive reserved for inserting into DATA pool vdevs when an active drive has failed."),
```

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 5]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
...
+special_vdev_title: T("Metadata"),
+special_vdev_description: T("Special Allocation class, used to create Fusion pools. Optional vdev type\
+ which is used to speed up metadata and small block IO. Cannot be removed from pool once added."),
+dedup_vdev_title: T("Dedup"),
+dedup_vdev_description: T("De-duplication Tables will be stored on this special vdev type, these will need\
+ to be sized to XGiB per XTiB of general storage.")
```
Comment (Mrt134) [inline src/app/helptext/storage/volumes/manager/manager.ts]:
```suggestion
dedup_vdev_description: T("De-duplication tables are stored on this special vdev type. These vdevs must\
 be sized to <i>X</i> GiB for each <i>X</i> TiB of general storage.")
```

File: src/app/helptext/storage/volumes/manager/manager.ts [hunk 6]
```diff
@@ -40,6 +40,20 @@ force_warning: T("The current pool layout is not recommended.\
...
+data_vdev_description: T("Normal vdev type, used for primary storage operations. A ZFS pool will always have at least one DATA vdev."),
+cache_vdev_title: T("Cache"),
+cache_vdev_description: T("ZFS L2ARC read-cache. Can be removed. Optional vdev that can be used with fast devices to accelerate read operations."),
+log_vdev_title: T("Log"),
+log_vdev_description: T("ZFS LOG device. Can be removed. Optional write-cache that can improve speeds of synchronous writes."),
+spare_vdev_title: T("Hot Spare"),
+spare_vdev_description: T("Drive set aside ready to be inserted into DATA pool vdevs when a device has failed."),
+special_vdev_title: T("Metadata"),
+special_vdev_description: T("Special Allocation class, used to create Fusion pools. Optional vdev type\
+ which is used to speed up metadata and small block IO. Cannot be removed from pool once added."),
```
Comment (william-gr) [inline src/app/helptext/storage/volumes/manager/manager.ts]: `Cannot be removed from pool once added.` is likely inaccurate. @amotin ?

**Comment by erinix** [comment]
> Is the pool status screen showing up OK with these two new vdevs and showing correct operations for them?

From what I have seen they look OK, they just show up like any other vdev would

File: src/app/helptext/storage/volumes/manager/manager.ts (hunk 6 above)
Comment (amotin) [reply src/app/helptext/storage/volumes/manager/manager.ts]: Yes, it is inaccurate. They can be removed.  But there are couple bugs in the area, that require more free space on remaining pool then really necessary for removal.  Hopefully users will have much more normal capacity then special to not hit this issue as I did on my tests.  There is a bug for that on OpenZFS github.

**Pull request merged**